src/main/java/seedu/address/model/person/Major.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a Person's major in the address book.
+ */
+public class Major {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Major is invalid. Please choose one of the following: ‘cs’, ‘bza’, ‘ceg’, ‘isys’, ‘isec’.";
+
+    /*
```

Comment:
Need to update the javadoc for this new validation regex. It accepts uppercase and lowercase and is case insensitive. Also, I think it would be good to handle leading and trailing whitespace, so the regex can be changed to `^[a-zA-Z]+$`.